front_end/src/components/landing_page/search_widget/SearchWidget.js

```diff
@@ -78,6 +82,27 @@ class SearchWidget extends Component {
       this.props.handleResetSearchOverview();
     }
 
+    //Check if the destination is given or not.
+    if (this.state.destinationName === "") {
+      this.setState({ showDesAlert: true });
+    } else {
+      this.setState({ showDesAlert: false });
+    }
+    //Check if the checkIn and checkOut date are given or not.
+    if (this.state.checkIn === "" || this.state.checkOut === "") {
+      this.setState({ showDateAlert: true });
+    } else {
+      this.setState({ showDateAlert: false });
+    }
+
+    if (
+      this.state.destinationName !== "" &&
```

Comment:
gotta check for null here too then.

OR

create a local variable within onSearchClick(), something like
validationError = false;

then you set the validationError to true if one of the inputs are missing... in line 98 you only need to check validationError, and not all the different things again.